arch/risc-v/src/mpfs/mpfs_irq.c

```diff
@@ -61,12 +74,6 @@ void up_irqinitialize(void)
   putreg32(0x0, iebase + 16);
   putreg32(0x0, iebase + 20);
 
-  /* Clear pendings in PLIC (for current hart) */
-
-  uintptr_t claim_address = mpfs_plic_get_claimbase();
-  uint32_t val = getreg32(claim_address);
-  putreg32(val, claim_address);
-
```

Comment:
Is this the faulty part in the code, should this not be a loop ?

  /* Clear pendings in PLIC (for current hart) */

```
  uintptr_t claim_address = mpfs_plic_get_claimbase();
  uint32_t val = getreg32(claim_address);
  while (val)
    {
      putreg32(val, claim_address);
      val = getreg32(claim_address);
    }
```
From the RISC-V privileged spec:
> The PLIC core will return an ID of zero, if there were no pending interrupts for the
> target when the claim was serviced.

Reply:
Ah, now I get it. The interrupt was already claimed, so this does nothing. What you did is better, it works in this insane corner case as well!

Reply:
Yes, claming anything here, or clearing pending interrupts, is not necessary. Only thing needed is to make sure that any IRQ which is already claimed gets also completed. Otherwise that IRQ will be "stuck", PLIC won't generate interrupts before the one is completed first. Also, any external interrupt can be active or can become active after clearing all the IRQ:s. So it is up to the interrupt source (device driver) initialization to manage startup of the device and enable its own IRQ properly.

Also, after the IE has been disabled it is no longer possible to claim that IRQ (reading the claim/complete will always return 0 after clearing all IEs). If one wanted to do clearing in the loop it should be done by using the threshold & interrupt pending bits with IE enabled, like this (from the spec):

"It is always legal for a hart to perform a claim even if the EIP is not set. In particular, a hart could set the threshold value to maximum to disable interrupt notifications and instead poll for active interrupts using periodic claim requests, though a simpler approach to implement polling would be to clear the external interrupt enable in the corresponding xie register for privilege mode x."

But again, I don't see why that would be necessary.

Reply:
I changed it according to your suggestion after all; it follows the existing style in the same file.